src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReportStatusCodeTest.java

```diff
@@ -0,0 +1,29 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+public class ReportStatusCodeTest {
+
+    @Test
+    void testFromCodeForValidReportStatusCodeReturnsCode() {
+        final ImmutableMap<String, ReportStatusCode> codeMap = ImmutableMap.<String, ReportStatusCode>builder()
+            .put("UN", ReportStatusCode.UNSPECIFIED)
+            .build();
+
+        assertEquals(ReportStatusCode.values().length, codeMap.size());
+        codeMap.forEach((code, statusCode) -> assertEquals(statusCode, ReportStatusCode.fromCode(code)));
+    }
```

Comment:
We can use the parameterized test with an enum source for this, unless you want the test to fail if/when we add new items to the enum.
```suggestion
    @ParameterizedTest
    @EnumSource(ReportStatusCode.class)
    void testFromCodeForValidReportStatusCodeReturnsCode(final ReportStatusCode reportStatusCode) {
        assertEquals(reportStatusCode, ReportStatusCode.fromCode(reportStatusCode.getCode()));
    }
```